ibis/backends/sql/compilers/base.py

```diff
@@ -109,21 +119,34 @@ def aggregate(
             Any arguments to pass to the aggregate.
         where
             An optional column filter to apply before performing the aggregate.
-
+        order_by
+            Optional ordering keys to use to order the rows before performing
+            the aggregate.
         """
         func = compiler.f[name]
 
-        if where is None:
-            return func(*args)
+        if order_by and not self.supports_order_by:
+            raise com.UnsupportedOperationError(
+                "ordering of order-sensitive aggregations via `order_by` is "
+                "not supported for this backend"
+            )
 
-        if self.supports_filter:
-            return sge.Filter(
-                this=func(*args),
-                expression=sge.Where(this=where),
+        if where is not None and not self.supports_filter:
+            args = tuple(
+                arg if isinstance(arg, sge.Literal) else compiler.if_(where, arg, NULL)
```

Comment:
I don't think ignoring `where` for all literals is correct.

There are other aggregate functions with more than one argument, e.g., `cov_samp(x, 2.4) filter (where y > 1)`. In that example `2.4` should be filtered out of the aggregate computation via a `NULL` and it wouldn't be in this implementation.

`group_concat` should be special cased and shouldn't go through the default `self.agg` mechanism, which I think was why it was that way in a few backends before.

Reply:
I'm not sure I follow this - if an aggregation takes multiple args and we only apply `where` to the non-literals (or even non-scalars), wouldn't null handling still work the same? Or even more concretely - do these not yield the same result?

```
cov_samp(iff(y > 1, x, NULL), 2.4)

cov_samp(iff(y > 1, x, NULL), iff(y > 1, 2.4, NULL)
```

Reply:
Assuming backends handle nulls the same in two argument functions, your argument works for the specific case of `cov_samp`, that is,

```sql
cov_sample(iff(y > 1, x, NULL), 2.4)
```

will evaluate to NULL, the same as

```sql
cov_sample(iff(y > 1, x, NULL), iff(y > 1, 2.4, NULL))
```

However, this doesn't generalize to arbitrary aggregates, because the actual input values change depending on the value of `y > 1`, so my argument is that we can't do this in general, because it assumes that all aggregates will always handle nulls the same way.